test/sql/aggregate/aggregates/test_mode.test

```diff
@@ -94,3 +94,82 @@ SELECT mode(i) FROM range(100) tbl(i) WHERE 1=0;
 ----
 NULL
 
+# Temporal types
```

Comment:
Maybe tests with groups and windows?

Reply:
I'll add some, but my longer term plan is to replace the current segment tree computation with a [moving mode](http://www.vldb.org/pvldb/vol9/p1221-wesley.pdf) algorithm that should use less memory at scale and probably be faster. I was going to add some nasty variable frame tests at that point, so I'll stick to basic correctness for now?

Reply:
Wow - `Combine` is destructive, which really doesn't play well with segment trees!

Reply:
And there is a bug in segment trees when the window size is 1 (the state never get constructed).